metamorphic: shared cache error on high-CPU machines

The number of secondary cache shards is a function of the number of CPUs, and the metamorphic tests use a fixed secondary cache size when enabled. On high-CPU machines, this can result in shared cache configuration validation erroring the metamorphic test out because the secondary cache size is too small for the configured number of shards.

```
        --- FAIL: TestMeta (0.02s)
            meta.go:471:
                        Error Trace:    meta.go:471
                                                                meta_test.go:66
                        Error:          Received unexpected error:
                                        pebble: could not open remote object cache: cache size 33554432 lower than min 50331648
                                        (1) attached stack trace
                                          -- stack trace:
                                          | github.com/cockroachdb/pebble/objstorage/objstorageprovider.(*provider).remoteInit
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/objstorage/objstorageprovider/remote.go:90
                                          | [...repeated from below...]
                                        Wraps: (2) pebble: could not open remote object cache
                                        Wraps: (3) attached stack trace
                                          -- stack trace:
                                          | github.com/cockroachdb/pebble/objstorage/objstorageprovider/sharedcache.Open
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/objstorage/objstorageprovider/sharedcache/shared_cache.go:135
                                          | github.com/cockroachdb/pebble/objstorage/objstorageprovider.(*provider).remoteInit
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/objstorage/objstorageprovider/remote.go:87
                                          | github.com/cockroachdb/pebble/objstorage/objstorageprovider.open
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/objstorage/objstorageprovider/provider.go:185
                                          | github.com/cockroachdb/pebble/objstorage/objstorageprovider.Open
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/objstorage/objstorageprovider/provider.go:149
                                          | github.com/cockroachdb/pebble.Open
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/open.go:318
                                          | github.com/cockroachdb/pebble/metamorphic.(*test).init.func12
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/metamorphic/test.go:133
                                          | github.com/cockroachdb/pebble/metamorphic.withRetries
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/metamorphic/retryable.go:18
                                          | github.com/cockroachdb/pebble/metamorphic.(*test).init
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/metamorphic/test.go:132
                                          | github.com/cockroachdb/pebble/metamorphic.RunOnce
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/metamorphic/meta.go:471
                                          | github.com/cockroachdb/pebble/internal/metamorphic.TestMeta
                                          |     /home/jackson/go/src/github.com/cockroachdb/pebble/internal/metamorphic/meta_test.go:66
                                          | testing.tRunner
                                          |     /snap/go/10339/src/testing/testing.go:1595
                                          | runtime.goexit
                                          |     /snap/go/10339/src/runtime/asm_amd64.s:1650
                                        Wraps: (4) cache size 33554432 lower than min 50331648
                                        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.leafError
                        Test:           TestMeta
```

cc @RaduBerinde, @joshimhoff

--- a/objstorage/objstorageprovider/sharedcache/shared_cache.go
+++ b/objstorage/objstorageprovider/sharedcache/shared_cache.go
@@ -130,9 +130,10 @@ func Open(
 	sizeBytes int64,
 	numShards int,
 ) (*Cache, error) {
-	min := shardingBlockSize * int64(numShards)
-	if sizeBytes < min {
-		return nil, errors.Errorf("cache size %d lower than min %d", sizeBytes, min)
+	if minSize := shardingBlockSize * int64(numShards); sizeBytes < minSize {
+		// Up the size so that we have one block per shard. In practice, this should
+		// only happen in tests.
+		sizeBytes = minSize
 	}
 
 	c := &Cache{

